src/render_cmd.rs

```diff
@@ -474,4 +486,185 @@ mod tests {
             res.pre_commands.get(PRE_CMD_DEPENDENCY_UPDATE).unwrap()
         );
     }
+
+    #[test]
+    fn filter_only_edge_deployment() {
+        let mut cfg = get_config();
+        cfg.chart = PathBuf::from("charts/some-chart");
+        cfg.namespace = Option::from("default".to_string());
+        cfg.release_name = "some-release".to_string();
+        cfg.output_path = PathBuf::from("manifests");
+
+        let mut edge_eu_w4_deployment = get_deployment();
+        let mut stage_eu_w4_deployment = get_deployment();
+        let mut prod_as_e1_deployment = get_deployment();
+        let mut prod_eu_w4_deployment = get_deployment();
+        let mut prod_us_c1_deployment = get_deployment();
+
+        edge_eu_w4_deployment.name = "edge_eu_w4_deployment".to_string();
+        stage_eu_w4_deployment.name = "stage_eu_w4_deployment".to_string();
+        prod_as_e1_deployment.name = "prod_as_e1_deployment".to_string();
+        prod_eu_w4_deployment.name = "prod_eu_w4_deployment".to_string();
+        prod_us_c1_deployment.name = "prod_us_c1_deployment".to_string();
+
+        cfg.deployments = vec![
+            edge_eu_w4_deployment,
+            stage_eu_w4_deployment,
+            prod_as_e1_deployment,
+            prod_eu_w4_deployment,
+            prod_us_c1_deployment,
+        ];
+
+        let mut cmd = get_cmd();
+        cmd.opts.filter = Option::from("edge".to_string());
+
+        let res = cmd.plan(&cfg).unwrap();
+        let expected_helm_cmd = "helm template some-release charts/some-chart --namespace=default --output-dir=manifests/edge_eu_w4_deployment/some-release";
+        let expected_helm_cmd: Vec<String> = expected_helm_cmd
+            .split_whitespace()
+            .map(String::from)
+            .collect();
+
+        assert_eq!(
```

Comment:
I would also suggest to assert that `res.commands`'s length is `1`, because we're only expecting a render command for edge.